File: template/src/main/java/ru/surfstudio/standard/ui/util/UiDateUtil.java
package ru.surfstudio.standard.ui.util;


import android.support.annotation.NonNull;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class UiDateUtil {

    private static final String ROLL_DATE_PATTERN = "HH:mm";
    private static final SimpleDateFormat sdf = new SimpleDateFormat(ROLL_DATE_PATTERN, Locale.getDefault());

    private UiDateUtil() {
    }

    /**
     * @return дата в формате "15:00"
     */
    public static String formatDateForDelivery(@NonNull Date date) {
        return sdf.format(date);
    }

    public static String formatDateForDeliveryEmptyIfNull(Date date) {
        return date != null ? formatDateForDelivery(date) : "";
    }
}

File: template/src/main/java/ru/surfstudio/standard/ui/util/UiDateUtil.kt
package ru.surfstudio.standard.ui.util


import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

object UiDateUtil {

    private val ROLL_DATE_PATTERN = "HH:mm"
    private val sdf = SimpleDateFormat(ROLL_DATE_PATTERN, Locale.getDefault())

    /**
     * @return дата в формате "15:00"
     */
    fun formatDateForDelivery(date: Date): String {
        return sdf.format(date)
    }

    fun formatDateForDeliveryEmptyIfNull(date: Date?): String {
        return if (date != null) formatDateForDelivery(date) else ""
    }
}
